kubemarine/patches/strong_ciphers_patch.py

```diff
@@ -0,0 +1,52 @@
+from textwrap import dedent
+import yaml
+from io import StringIO
+from typing import Any, Dict, List
+
+from kubemarine.core.action import Action
+from kubemarine.core.patch import RegularPatch
+from kubemarine.core.resources import DynamicResources
+from kubemarine import kubernetes
+
+class TheAction(Action):
+    def __init__(self) -> None:
+        super().__init__("Update API Server TLS cipher suites")
+
+    def run(self, res: DynamicResources) -> None:
+        cluster = res.cluster()
+        kubernetes_nodes = cluster.make_group_from_roles(['control-plane'])
+
+        for member_node in kubernetes_nodes.get_ordered_members_list():
+            apiserver_file = "/etc/kubernetes/manifests/kube-apiserver.yaml"
+
+            # Load the YAML configuration
+            try:
+                apiserver_config = yaml.safe_load(member_node.sudo(f"cat {apiserver_file}").get_simple_out())
+            except yaml.YAMLError as exc:
+                cluster.log.error(f"Failed to parse YAML file: {exc}")
+                return
+
+            # Modify the YAML structure (adjust the path as needed)
+            apiserver_config['spec']['containers'][0]['command'].append("--tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384")
```

Comment:
What if the manifest already has the `--tls-cipher-suite` arg? For example, someone installed the cluster using this guide https://github.com/Netcracker/KubeMarine/blob/main/documentation/internal/Hardening.md#automated-application-for-strong-cyrptographic-ciphers-for-api-server-during-new-cluster-installation

Also, probably the guide `Use strong cryptographic ciphers for API server` becomes no longer actual because we are starting to ensure corresponding parameters.

Reply:
Added a logic to check if it already has the --tls-cipher-suite before applying patch